client/packages/programs/src/JsonForms/components/IdGenerator.tsx

```diff
@@ -140,19 +144,28 @@ const Part: z.ZodType<Part> = z.discriminatedUnion('type', [
   }),
 ]);
 
-const IdValidation: z.ZodType<IdValidation> = z
-  .object({
-    type: z.literal('UniqueEnrolmentId'),
-  })
-  .strict();
+const EnrolmentIdValidation = z.object({
+  type: z.literal('UniqueEnrolmentId'),
+});
+
+const PatientCodeValidation = z.object({
+  type: z.literal('UniquePatientCode'),
+});
 
 const GeneratorOptions: z.ZodType<GeneratorOptions> = z
   .object({
     parts: z.array(Part),
     confirmRegenerate: z.boolean().optional().default(true),
     preventRegenerate: z.boolean().optional().default(false),
     allowManualEntry: z.boolean().optional(),
-    validations: z.array(IdValidation).optional(),
+    validations: z
+      .array(
+        z.discriminatedUnion('type', [
+          EnrolmentIdValidation,
+          PatientCodeValidation,
+        ])
+      )
+      .optional(),
```

Comment:
Just a note, in later components we only use zod without defining an extra interface. You can get the interface directly from zod...

Reply:
I wonder would you be willing to send an example component where this is called directly?

Reply:
`packages/programs/src/JsonForms/components/AdherenceScore.tsx`

The nice thing is that the docs comments on the zod definitions are still working when using the parsed options :)